test/bin.js

```diff
@@ -0,0 +1,203 @@
+  mkdirp.sync(dir)
+  fs.writeFileSync(join(dir, '.testrc'), JSON.stringify({
+    host: '127.0.0.1',
+    port: 8399,
+    ws: {
+      port: 8499
+    }
+  }))
+  var sbot = spawn(join(__dirname, '../bin.js'),  [
+    'server',
+    `--path=${dir}`
+  ], {
+    env: Object.assign({}, process.env, {ssb_appname: 'test'}),
+    cwd: dir,
+    stdio: 'inherit'
+  })
+
+  sbot.on('exit', function(code) {
+    t.notOk(code, 'sbot exit code should be zero')
+    t.end()
+  })
+
+  setTimeout( function() {
+    var socket = net.connect(8399, '127.0.0.1')
+    socket.on('error', function(err) {
+      t.fail(err)
+      sbot.kill('SIGINT')
+    })
+    socket.on('connect', function(err) {
+      t.pass('connected successfully')
+      sbot.kill('SIGINT')
+    })
+  }, 1200)
+})
+
+if (has_ipv6)
+test('run bin.js server with local config file (port, host) (IPv6)', function(t) {
+  var dir = '/tmp/' + Date.now()
+  mkdirp.sync(dir)
+  fs.writeFileSync(join(dir, '.testrc'), JSON.stringify({
+    host: '::',
+    port: 8399,
+    ws: {
+      port: 8499
+    }
+  }))
+  var sbot = spawn(join(__dirname, '../bin.js'),  [
+    'server',
+    `--path=${dir}`
+  ], {
+    env: Object.assign({}, process.env, {ssb_appname: 'test'}),
+    cwd: dir,
+    stdio: 'inherit'
+  })
+
+  sbot.on('exit', function(code) {
+    t.notOk(code, 'sbot exit code should be zero')
+    t.end()
+  })
+
+  setTimeout( function() {
+    var socket = net.connect(8399, '::1')
+    socket.on('error', function(err) {
+      t.fail(err)
+      sbot.kill('SIGINT')
+    })
+    socket.on('connect', function(err) {
+      t.pass('connected successfully')
+      sbot.kill('SIGINT')
+    })
+  }, 1200)
+})
+
+test('sbot should have websockets and http server by default', function(t) {
+  var path = '/tmp/' + Date.now()
+  var killed = false
+  var sbot = spawn(join(__dirname, '../bin.js'),  [
+    'server',
+    '--host=127.0.0.1',
+    '--port=8199',
+    '--ws.port=0', // use random port - get them from getAddress and dial that
+    '--path=' + path
+  ], {
+    env: Object.assign({}, process.env, {ssb_appname: 'test'}),
+    stdio: 'inherit'
+  })
+
+  sbot.on('exit', function(code) {
+    if (!killed) t.notOk(code, 'sbot exit code should be zero')
+    t.end()
+  })
+
+  setTimeout( function() {
+    exec([
+      join(__dirname, '../bin.js'),
+      'getAddress',
+      '--',
+      '--host=127.0.0.1',
+      '--port=8199',
+      '--path=' + path
+    ].join(' '), {
+      env: Object.assign({}, process.env, {ssb_appname: 'test'})
+    }, function(err, stdout, sderr) {
+      t.error(err)
+      console.log('result of sbot getAddress', stdout)
+      const idx = stdout.indexOf('ws:')
+      t.ok(idx > 0, 'multiserver address should contain ws:')
+
+      require('ssb-client')(null, { // falsy value => load default from path
+        path: path,
+        caps: { shs: require('../lib/ssb-cap') }, // has to be set when setting any config
+        remote: stdout.substr(idx)
```

Comment:
this relies on the ws address to be the last one  within the multiserver address, that's not good.

Reply:
I see. We probably want to test both ws.getAddress and getAddress.